File: solutions/lua/allergies/1/allergies.lua
local function list(score)
  score = score % 256
  local scores = {
    { "cats", 128 },
    { "pollen", 64 },
    { "chocolate", 32 },
    { "tomatoes", 16 },
    { "strawberries", 8 },
    { "shellfish", 4 },
    { "peanuts", 2 },
    { "eggs", 1 }
  }
  local allergies = {}
  for _, v in ipairs(scores) do
    if score >= v[2] then
      table.insert(allergies, 1, v[1])
      score = score - v[2]
    end
  end
  return allergies
end

local function allergic_to(score, which)
  local x = false
  for _, v in ipairs(list(score)) do
    if v == which then x = true end
  end
  return score < 0 or x
end

return { list = list, allergic_to = allergic_to }

File: solutions/go/allergies/1/allergies.go
package allergies

var list = []struct {
	score uint
	name  string
}{
	{128, "cats"},
	{64, "pollen"},
	{32, "chocolate"},
	{16, "tomatoes"},
	{8, "strawberries"},
	{4, "shellfish"},
	{2, "peanuts"},
	{1, "eggs"},
}

func Allergies(allergies uint) []string {
	var res []string
	for _, item := range list {
		if allergies&item.score != 0 {
			res = append(res, item.name)
		}
	}
	return res
}

func AllergicTo(allergies uint, allergen string) bool {
	for _, item := range list {
		if item.name == allergen {
			if allergies&item.score != 0 {
				return true
			}
			return false
		}
	}
	return false
}
